File: app/frontend/src/core/interfaces/types/monster.ts
// Monster siege and attack types

export type MonsterAttackDirection =
  | 'north'
  | 'northeast'
  | 'southeast'
  | 'south'
  | 'southwest'
  | 'northwest';

export type MonsterAttackOutcome =
  | 'defended'
  | 'close_defeat'
  | 'decisive_defeat';

export interface MonsterAttack {
  id: string;
  poiId: string;
  direction: MonsterAttackDirection;
  strength: number;
  monsterTypes: string[];
  timestamp: Date;
  resolved: boolean;
  outcome?: MonsterAttackOutcome;
} 

File: backend/app/src/core/interfaces/types/monster.py
from typing import Any, List, Union


MonsterAttackDirection = Union[, 'north', 'northeast', 'southeast', 'south', 'southwest', 'northwest']
MonsterAttackOutcome = Union[, 'defended', 'close_defeat', 'decisive_defeat']
class MonsterAttack:
    id: str
    poiId: str
    direction: MonsterAttackDirection
    strength: float
    monsterTypes: List[str]
    timestamp: Date
    resolved: bool
    outcome?: MonsterAttackOutcome 